File: teamplaner/www/teamverwaltung.py
# -*- coding: utf-8 -*-
# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
# MIT License. See license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
from teamplaner.teamplaner.doctype.teamplaner_mitglied.teamplaner_mitglied import invite_user, add_user_to_events
from teamplaner.www.statistik import get_saisondaten

#no_cache = 1

def get_context(context):
	if frappe.session.user=='Guest':
		frappe.throw(_("You need to be logged in to access this page"), frappe.PermissionError)
	if "TeamPlaner Trainer" not in frappe.get_roles(frappe.session.user):
		frappe.throw(_("Du benötigst eine Trainer Rolle für den Zugriff auf diese Seite"), frappe.PermissionError)
	context.show_sidebar=True
	
	context['spieler'] = frappe.db.sql("""SELECT `vorname`, `nachname`, `name` FROM `tabTeamPlaner Mitglied`""", as_dict=True)
	context['trainings'] = frappe.db.sql("""SELECT `name`, `ort` FROM `tabTeamPlaner Training`""", as_dict=True)
	context['spiele'] = frappe.db.sql("""SELECT `name`, `ort` FROM `tabTeamPlaner Spiel`""", as_dict=True)
	context['saisondaten'] = get_saisondaten()
	return context
	
@frappe.whitelist()
def neuer_spieler(vorname, nachname, mail, lizenznummer, nummer, position, linie):
	spieler = frappe.new_doc("TeamPlaner Mitglied")
	spieler.vorname = vorname
	spieler.nachname = nachname
	spieler.mail = mail
	spieler.lizenznummer = lizenznummer
	spieler.rueckennummer = nummer
	spieler.position = position
	spieler.linie = linie
	team = spieler.append('belongs_to_team', {})
	team.team = "Herren 2"
	team.spieler = 1
	spieler.save()
	invite_user(spieler.name)
	add_user_to_events(spieler.name)
	return spieler.name
	
@frappe.whitelist()
def get_spieler_details(spieler):
	spieler = frappe.get_doc("TeamPlaner Mitglied", spieler)
	return spieler
	
@frappe.whitelist()
def spieler_bearbeiten(spieler, vorname, nachname, lizenznummer, nummer, position, linie):
	spieler = frappe.get_doc("TeamPlaner Mitglied", spieler)
	spieler.vorname = vorname
	spieler.nachname = nachname
	spieler.lizenznummer = lizenznummer
	spieler.rueckennummer = nummer
	spieler.position = position
	spieler.linie = linie
	spieler.save()
	return spieler.name
	
@frappe.whitelist()
def spieler_entfernen(spieler):
	spieler = frappe.get_doc("TeamPlaner Mitglied", spieler)
	spieler.delete()
	return
	
@frappe.whitelist()
def neues_training(wann, wo, von, bis, details):
	training = frappe.new_doc("TeamPlaner Training")
	training.datum = wann
	training.ort = wo
	training.von = str(von) + ":00"
	training.bis = str(bis) + ":00"
	training.team = "Herren 2"
	training.details = details
	training.save()
	return training.name
	
@frappe.whitelist()
def get_training_details(training):
	training = frappe.get_doc("TeamPlaner Training", training)
	return training
	
@frappe.whitelist()
def training_bearbeiten(training, wo, von, bis, details):
	training = frappe.get_doc("TeamPlaner Training", training)
	training.ort = wo
	training.von = von
	training.bis = bis
	training.details = details
	training.save()
	return training.name
	
@frappe.whitelist()
def training_entfernen(training):
	training = frappe.get_doc("TeamPlaner Training", training)
	training.delete()
	return
	
@frappe.whitelist()
def neues_spiel(wann, wo, von, bis, gegner_1, zwei_spiele, gegner_2, eins_heimspiel, zwei_heimspiel):
	spiel = frappe.new_doc("TeamPlaner Spiel")
	spiel.datum = wann
	spiel.ort = wo
	spiel.von = str(von) + ":00"
	spiel.bis = str(bis) + ":00"
	spiel.team = "Herren 2"
	spiel.gegner = gegner_1
	spiel.zweiter_gegner = gegner_2
	spiel.zwei_spiele = zwei_spiele
	spiel.eins_heimspiel = eins_heimspiel
	spiel.zwei_heimspiel = zwei_heimspiel
	
	spiel.save()
	return spiel.name
	
@frappe.whitelist()
def get_spiel_details(spiel):
	spiel = frappe.get_doc("TeamPlaner Spiel", spiel)
	return spiel
	
@frappe.whitelist()
def spiel_entfernen(spiel):
	spiel = frappe.get_doc("TeamPlaner Spiel", spiel)
	spiel.delete()
	return
	
@frappe.whitelist()
def delete_old_data():
	training = frappe.db.sql("""DELETE FROM `tabTeamPlaner Training` WHERE `name` < CURDATE()""", as_dict=True)
	spiele = frappe.db.sql("""DELETE FROM `tabTeamPlaner Spiel` WHERE `name` < CURDATE()""", as_dict=True)
	return
	
@frappe.whitelist()
def change_saisondaten(von, bis):
	team = frappe.get_doc("TeamPlaner Team", "Herren 2")
	team.saison_von = von
	team.saison_bis = bis
	team.save()
	return

File: teamplaner/www/teamverwaltung.js
function show_neuer_spieler() {
	document.getElementById("team_section").style.display = "none";
	document.getElementById("neuer_spieler").style.display = "block";
}

function show_spieler_bearbeiten() {
	document.getElementById("team_section").style.display = "none";
	document.getElementById("spieler_bearbeiten").style.display = "block";
}

function show_allgemeine_einstellungen() {
	document.getElementById("saison_von").value = "{{ saisondaten.saison_von }}";
	document.getElementById("saison_bis").value = "{{ saisondaten.saison_bis }}";
	document.getElementById("team_section").style.display = "none";
	document.getElementById("allgemeine_einstellungen").style.display = "block";
}

function reset_tab_team() {
	document.getElementById("team_section").style.display = "block";
	document.getElementById("spieler_bearbeiten").style.display = "none";
	document.getElementById("neuer_spieler").style.display = "none";
	document.getElementById("allgemeine_einstellungen").style.display = "none";
}

function show_neues_training() {
	document.getElementById("training_section").style.display = "none";
	document.getElementById("neues_training").style.display = "block";
}

function show_training_bearbeiten() {
	document.getElementById("training_section").style.display = "none";
	document.getElementById("training_bearbeiten").style.display = "block";
}

function reset_tab_training() {
	document.getElementById("training_section").style.display = "block";
	document.getElementById("neues_training").style.display = "none";
	document.getElementById("training_bearbeiten").style.display = "none";
}

function show_neues_spiel() {
	document.getElementById("spiel_section").style.display = "none";
	document.getElementById("neues_spiel").style.display = "block";
}

function show_spiel_bearbeiten() {
	document.getElementById("spiel_section").style.display = "none";
	document.getElementById("spiel_bearbeiten").style.display = "block";
}

function reset_tab_spiel() {
	document.getElementById("spiel_section").style.display = "block";
	document.getElementById("neues_spiel").style.display = "none";
	document.getElementById("spiel_bearbeiten").style.display = "none";
}

function add_spieler() {
	frappe.show_message("Bitte warten");
	var vorname = document.getElementById("vorname").value;
	var nachname = document.getElementById("nachname").value;
	var mail = document.getElementById("mail").value;
	var lizenznummer = document.getElementById("lizenznummer").value;
	var nummer = document.getElementById("nummer").value;
	var position = document.getElementById("position").value;
	var linie = document.getElementById("linie").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.neuer_spieler",
		'args': {
			'vorname': vorname,
			'nachname': nachname,
			'mail': mail,
			'lizenznummer': lizenznummer,
			'nummer': nummer,
			'position': position,
			'linie': linie
		},
		'callback': function(r) {
			document.getElementById("vorname").value = "";
			document.getElementById("nachname").value = "";
			document.getElementById("mail").value = "";
			document.getElementById("lizenznummer").value = "";
			document.getElementById("nummer").value = "";
			document.getElementById("position").value = "";
			document.getElementById("linie").value = "";
			frappe.hide_message();
			frappe.msgprint("Es wurde erfolgreich ein neuer Spieler angelegt.");
		}
	});
}

function get_spieler_details() {
	var spieler = document.getElementById("spieler").value;
	if (spieler != "Bitte wähle einen Spieler aus") {
		frappe.call({
			'method': "teamplaner.www.teamverwaltung.get_spieler_details",
			'args': {
				'spieler': spieler
			},
			'callback': function(r) {
				var spieler = r.message;
				document.getElementById("bea_vorname").value = spieler.vorname;
				document.getElementById("bea_nachname").value = spieler.nachname;
				document.getElementById("bea_lizenznummer").value = spieler.lizenznummer;
				document.getElementById("bea_nummer").value = spieler.rueckennummer;
				document.getElementById("bea_position").value = spieler.position;
				document.getElementById("bea_linie").value = spieler.linie;
			}
		});
	} else {
		document.getElementById("bea_vorname").value = "";
		document.getElementById("bea_nachname").value = "";
		document.getElementById("bea_lizenznummer").value = "";
		document.getElementById("bea_nummer").value = "";
		document.getElementById("bea_position").value = "";
		document.getElementById("bea_linie").value = "";
	}
}

function add_training() {
	frappe.show_message("Bitte warten");
	var wann = document.getElementById("wann").value;
	var wo = document.getElementById("wo").value;
	var von = document.getElementById("von").value;
	var bis = document.getElementById("bis").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.neues_training",
		'args': {
			'wann': wann,
			'wo': wo,
			'von': von,
			'bis':bis
		},
		'callback': function(r) {
			frappe.hide_message();
			frappe.msgprint("Das Training wurde erfolgreich angelegt.");
		}
	});
}

function get_training_details() {
	var training = document.getElementById("training").value;
	if (training != "Bitte wähle ein Training aus") {
		frappe.call({
			'method': "teamplaner.www.teamverwaltung.get_training_details",
			'args': {
				'training': training
			},
			'callback': function(r) {
				var training = r.message;
				document.getElementById("bea_wann").value = training.datum;
				document.getElementById("bea_wo").value = training.ort;
				document.getElementById("bea_von").value = training.von;
				document.getElementById("bea_bis").value = training.bis;
			}
		});
	} else {
		document.getElementById("bea_wann").value = "";
		document.getElementById("bea_wo").value = "";
		document.getElementById("bea_von").value = "";
		document.getElementById("bea_bis").value = "";
	}
}

function add_spiel() {
	frappe.show_message("Bitte warten");
	var wann = document.getElementById("spiel_wann").value;
	var wo = document.getElementById("spiel_wo").value;
	var von = document.getElementById("spiel_von").value;
	var bis = document.getElementById("spiel_bis").value;
	if (document.getElementById("1spiel").checked) {
		var zwei_spiele = 0;
		var zwei_heimspiel = 0;
		var gegner_1 = document.getElementById("gegner_1").value;
		var gegner_2 = document.getElementById("gegner_2").value;
		if (document.getElementById("1heim").checked) {
			var eins_heimspiel = 1;
		} else {
			var eins_heimspiel = 0;
		}
	} else{
		var zwei_spiele = 1;
		var gegner_1 = document.getElementById("gegner_1").value;
		var gegner_2 = document.getElementById("gegner_2").value;
		if (document.getElementById("1heim").checked) {
			var eins_heimspiel = 1;
		} else {
			var eins_heimspiel = 0;
		}
		if (document.getElementById("2heim").checked) {
			var zwei_heimspiel = 1;
		} else {
			var zwei_heimspiel = 0;
		}
	}
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.neues_spiel",
		'args': {
			'wann': wann,
			'wo': wo,
			'von': von,
			'bis':bis,
			'gegner_1': gegner_1,
			'zwei_spiele': zwei_spiele,
			'gegner_2': gegner_2,
			'eins_heimspiel': eins_heimspiel,
			'zwei_heimspiel': zwei_heimspiel
		},
		'callback': function(r) {
			frappe.hide_message();
			document.getElementById("spiel_wann").value = "";
			document.getElementById("spiel_wo").value = "";
			document.getElementById("spiel_von").value = "";
			document.getElementById("spiel_bis").value = "";
			document.getElementById("gegner_1").value = "";
			document.getElementById("gegner_2").value = "";
			document.getElementById("1heim").checked = false;
			document.getElementById("2heim").checked = false;
			document.getElementById("2spiele").checked = true;
			frappe.msgprint("Das Spiel wurde erfolgreich angelegt.");
		}
	});
}

function change_spieler() {
	frappe.show_message("Bitte warten");
	var spieler = document.getElementById("spieler").value;
	var vorname = document.getElementById("bea_vorname").value;
	var nachname = document.getElementById("bea_nachname").value;
	var lizenznummer = document.getElementById("bea_lizenznummer").value;
	var nummer = document.getElementById("bea_nummer").value;
	var position = document.getElementById("bea_position").value;
	var linie = document.getElementById("bea_linie").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.spieler_bearbeiten",
		'args': {
			'spieler': spieler,
			'vorname': vorname,
			'nachname': nachname,
			'lizenznummer': lizenznummer,
			'nummer': nummer,
			'position': position,
			'linie': linie
		},
		'callback': function(r) {
			frappe.hide_message();
			frappe.msgprint("Der Spieler wurde erfolgreich geändert.");
		}
	});
}

function change_training() {
	frappe.show_message("Bitte warten");
	var training = document.getElementById("training").value;
	var wo = document.getElementById("bea_wo").value;
	var von = document.getElementById("bea_von").value;
	var bis = document.getElementById("bea_bis").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.training_bearbeiten",
		'args': {
			'training': training,
			'wo': wo,
			'von': von,
			'bis':bis
		},
		'callback': function(r) {
			frappe.hide_message();
			frappe.msgprint("Das Training wurde erfolgreich geändert.");
		}
	});
}

function change_spiel() {
	
}

function delete_spieler() {
	frappe.show_message("Bitte warten");
	var spieler = document.getElementById("spieler").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.spieler_entfernen",
		'args': {
			'spieler': spieler
		},
		'callback': function(r) {
			frappe.hide_message();
			document.getElementById("spieler").value = "Bitte wähle einen Spieler aus";
			document.getElementById("bea_vorname").value = "";
			document.getElementById("bea_nachname").value = "";
			document.getElementById("bea_lizenznummer").value = "";
			document.getElementById("bea_nummer").value = "";
			document.getElementById("bea_position").value = "";
			document.getElementById("bea_linie").value = "";
			frappe.msgprint("Der Spieler wurde erfolgreich gelöscht.");
		}
	});
}

function delete_training() {
	frappe.show_message("Bitte warten");
	var training = document.getElementById("training").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.training_entfernen",
		'args': {
			'training': training
		},
		'callback': function(r) {
			frappe.hide_message();
			document.getElementById("training").value = "Bitte wähle ein Training aus";
			document.getElementById("bea_wo").value = "";
			document.getElementById("bea_von").value = "";
			document.getElementById("bea_bis").value = "";
			frappe.msgprint("Das Training wurde erfolgreich gelöscht.");
		}
	});
}

function delete_spiel() {
	frappe.show_message("Bitte warten");
	var spiel = document.getElementById("spiel").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.spiel_entfernen",
		'args': {
			'spiel': spiel
		},
		'callback': function(r) {
			frappe.hide_message();
			document.getElementById("spiel").value = "Bitte wähle ein Spiel aus";
			document.getElementById("bea_spiel_wann").value = "";
			document.getElementById("bea_spiel_wo").value = "";
			document.getElementById("bea_spiel_von").value = "";
			document.getElementById("bea_spiel_bis").value = "";
			document.getElementById("bea_gegner_1").value = "";
			document.getElementById("bea_gegner_2").value = "";
			document.getElementById("bea_1heim").checked = false;
			document.getElementById("bea_2heim").checked = false;
			document.getElementById("bea_1spiel").checked = false;
			document.getElementById("bea_2spiele").checked = false;
			frappe.msgprint("Das Spiel wurde erfolgreich gelöscht.");
		}
	});
}

function change_anz_spiele(anz) {
	if (anz == '1') {
		document.getElementById("gegner_2").disabled = true;
		document.getElementById("2heim").disabled = true;
	} else {
		document.getElementById("gegner_2").disabled = false;
		document.getElementById("2heim").disabled = false;
	}
}

function bea_change_anz_spiele(anz) {
	if (anz == '1') {
		document.getElementById("bea_gegner_2").disabled = true;
		document.getElementById("bea_2heim").disabled = true;
	} else {
		document.getElementById("bea_gegner_2").disabled = false;
		document.getElementById("bea_2heim").disabled = false;
	}
}

function get_spiel_details() {
	var spiel = document.getElementById("spiel").value;
	if (spiel != "Bitte wähle ein Spiel aus") {
		frappe.call({
			'method': "teamplaner.www.teamverwaltung.get_spiel_details",
			'args': {
				'spiel': spiel
			},
			'callback': function(r) {
				var spiel = r.message;
				document.getElementById("bea_spiel_wann").value = spiel.datum;
				document.getElementById("bea_spiel_wo").value = spiel.ort;
				document.getElementById("bea_spiel_von").value = spiel.von;
				document.getElementById("bea_spiel_bis").value = spiel.bis;
				document.getElementById("bea_gegner_1").value = spiel.gegner;
				document.getElementById("bea_gegner_2").value = spiel.zweiter_gegner;
				if (spiel.eins_heimspiel == 1) {
					document.getElementById("bea_1heim").checked = true;
				} else {
					document.getElementById("bea_1heim").checked = false;
				}
				if (spiel.zwei_heimspiel == 1) {
					document.getElementById("bea_2heim").checked = true;
				} else {
					document.getElementById("bea_2heim").checked = false;
				}
				if (spiel.zwei_spiele == 1) {
					document.getElementById("bea_2spiele").checked = true;
				} else {
					document.getElementById("bea_1spiel").checked = true;
				}
			}
		});
	} else {
		document.getElementById("bea_spiel_wann").value = "";
		document.getElementById("bea_spiel_wo").value = "";
		document.getElementById("bea_spiel_von").value = "";
		document.getElementById("bea_spiel_bis").value = "";
		document.getElementById("bea_gegner_1").value = "";
		document.getElementById("bea_gegner_2").value = "";
		document.getElementById("bea_1heim").checked = false;
		document.getElementById("bea_2heim").checked = false;
		document.getElementById("bea_1spiel").checked = false;
		document.getElementById("bea_2spiele").checked = false;
	}
}

function delete_alte_daten() {
	frappe.show_message("Bitte warten");
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.delete_old_data",
		'args': {},
		'callback': function(r) {
			frappe.hide_message();
			frappe.msgprint("Alle Trainings/Spiele vor dem heutigen Datum wurden entfernt.");
		}
	});
}

function allgemeine_einstellungen_save() {
	frappe.show_message("Bitte warten");
	var von = document.getElementById("saison_von").value;
	var bis = document.getElementById("saison_bis").value;
	frappe.call({
		'method': "teamplaner.www.teamverwaltung.change_saisondaten",
		'args': {
			'von': von,
			'bis': bis
		},
		'callback': function(r) {
			frappe.hide_message();
			frappe.msgprint("Die Saisondaten wurden angepasst.");
		}
	});
}